circuits/cpp/scripts/run_coverage

```diff
@@ -0,0 +1,78 @@
+#!bin/bash
+set -e
+
+# To be called LOCALLY for testing WITHOUT docker.
+#
+# Contains core logic for collecting test coverage in the current environment.
+#
+# Run from circuits/cpp/
+# Example:
+# ./scripts/run_coverage glob
+#
+# Running with the CLEAN environment variable set will delete the build-coverage folder and start again.
+# Running with the CLEAR_COV environment variable set will clear existing vscode coverage data (lcov.info).
+#
+# This will create two outputs: lcov.info and an html coverage report in build-coverage/coverage_report
+#
+# Note: Assumes ignition SRS transcripts have already been downloaded
+
+# TESTS:
+# 1. names of test executables to run
+#   - (like "aztec3_circuits_kernel_tests aztec3_circuits_abis_tests")
+# 2. or "glob" (LOWERCASE) to run all built tests
+# 3. or a file containing a list of test
+TESTS=$1
+
+# Set coverage build directory
+BUILD_DIR="build-coverage"
+
+# If the clean file is set, wipe all existing coverage data
+if [[ -n "$CLEAN" ]]; then
+  echo "The CLEAN environment variable is set. Clearing existing coverage data."
+  rm -rf ./$BUILD_DIR || true
+fi
+
+# Perform coverage build
+echo "Setting cmake to run with preset coverage"
+cmake -Bbuild-coverage -DCMAKE_BUILD_TYPE=Debug -DUSE_TURBO=true --preset=coverage .
+
+# if TESTS is GLOB or empty, run all built tests
+# if TESTS is a file, assume it contains a list of tests to run
+# otherwise assume TESTS itself is a list of tests to run
+if [ "$TESTS" == "glob" ] || [ -z "$TESTS" ]; then
+  echo "    No test file specified. Building all test files."
+  TESTS=$(cat ./scripts/a3-tests)
+elif [ -f "$TESTS" ]; then
+  TESTS=$(cat $TESTS | tr '\n' ' ')
+fi
+
+cd $BUILD_DIR
+
+# if just the CLEAR_COV env var is set then we will clear existing coveage report information
+if [[ -n "$CLEAR_COV" ]]; then
+  echo "The CLEAR_COV environment variable is set. Clearing existing coverage data."
+  rm ./lcov.info || true
+  rm ./merged_profdata/default.profdata || true
+  rm ./default.profraw || true
+  rm -rf ./bin/profdata || true
+fi
+
+for BIN in $TESTS; do
+  echo "*** Building $BIN ***"
+  cmake --build . --target $BIN
+done
+
+echo "Testing $BIN"
+echo "*** Collecting Profiling Information ***"
+ctest || true
```

Comment:
Why do we allow this to fail? At the very least maybe it should be `ctest || echo "CTest profiling failed, <some other info>"

Reply:
When running for only one test binary (`run_coverage aztec3_circuits_rollup_tests`), it will print [NOT BUILT] on the other test binaries and return 1, preventing the rest of the script to run. 
`|| true` was just a way to allow it to continue if certain tests hadn't been built. I couldn't find a ctest flag that prevented it exiting in this case.

Reply:
Okay I see. Since `ctest` itself outputs the warnings this is probably fine. If there is some way for us to only run the test exe we care about that would be better since this will run all tests otherwise. But if that is too much effort then this is fine.

Add a comment here saying "ctest will fail if there are unbuilt tests".